src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/ComService.kt

```diff
@@ -129,4 +134,77 @@ class ComService(
       onProbationCount,
     )
   }
+
+  private fun PrisonerSearchPrisoner.getLicenceType(): LicenceType {
+    return if (this.licenceExpiryDate.isNullOrBlank()) {
+      LicenceType.PSS
+    } else if (this.topUpSupervisionExpiryDate.isNullOrBlank() || LocalDate.parse(
+        this.topUpSupervisionExpiryDate,
+      ) <= LocalDate.parse(this.licenceExpiryDate)
+    ) {
+      LicenceType.AP
+    } else {
+      LicenceType.AP_PSS
+    }
+  }
```

Comment:
this `if return / else if return / else return` is often much nicer handled by pattern matching.

With date parsing pushed to the boundary, it could look something like:

```
  private fun PrisonerSearchPrisoner.getLicenceType() = when {
    licenceExpiryDate == null -> PSS
    topUpSupervisionExpiryDate == null || topUpSupervisionExpiryDate <= licenceExpiryDate -> AP
    else -> AP_PSS
  }
```

Reply:
Done